packages/idyll-cli/bin/cmds/create-project.js

```diff
@@ -0,0 +1,175 @@
+    });
+    questions.push({
+      name: 'template',
+      type: 'list',
+      message: 'Which project template would you like to use?',
+      choices: templates
+    });
+    return inquirer.prompt(questions);
+  }
+
+  async function ensureDefaults (answers) {
+    if (!answers['project-dir']) answers['project-dir'] = projectDir;
+    return answers;
+  }
+}
+
+async function createProject (answers) {
+  let name = answers['package-name'];
+  let template = answers['template'];
+  let dir = answers['project-dir'];
+
+  let startMessage = `Creating a new Idyll project in ${dir} using the ${template} template...`
+  let successMessage = 'Finished creating the project!';
+  let doneInstructions = `To start developing, run the following commands in your terminal:
+    cd ${dir}
+    idyll
+  `;
+  let dirExistsMessage = 'That directory already exists. Please ensure that your target directory\
+\ does not exist.';
+  let errorMessage = `Could not create Idyll project in ${dir}`;
+
+  let stages = [
+    ['Ensuring that the target directory is valid', ensureEmptyProject],
+    ['Copying files from template directory into the target directory', copyFiles],
+    ['Configuring project', fillTemplates],
+    ['Installing dependencies', installDependencies]
+  ];
+
+  console.log(colors.success(startMessage));
+  var spinner
+  try {
+    for (var i = 0; i < stages.length; i++) {
+      let stage = stages[i]
+      spinner = ora({
+        text: colors.progress(' ' + stage[0])
+      });
+      spinner.start();
+      await stage[1]();
+      spinner.succeed()
+    }
+  } catch (err) {
+    if (spinner) spinner.fail(colors.failure(err));
+    return;
+  }
+  console.log();
+  console.log(colors.success(successMessage));
+  console.log(colors.progress(doneInstructions));
+
+  async function ensureEmptyProject () {
+    return fs.pathExists(dir)
+      .then(exists => {
+        if (exists) {
+          throw new Error(dirExistsMessage);
+        }
+        return true;
+      });
+  }
+
+  async function copyFiles (proceed) {
+    await fs.copy(getTemplatePath(template), dir);
+    await fs.copy(DEFAULT_COMPONENTS_DIR, p.join(dir, 'components', 'default'));
+  }
+
+  async function fillTemplates () {
+    let packagePath = p.join(dir, 'package.json');
+    let indexPath = p.join(dir, 'index.idyll');
+
+    let packageJson = JSON.parse(await fs.readFile(packagePath));
+    let indexIdyll = await fs.readFile(indexPath, { encoding: 'utf-8' });
+
+    packageJson.name = name;
+    var title = name.split('-').join(' ').replace(/\b\w/g, l => l.toUpperCase())
+
+    await fs.writeFile(packagePath, JSON.stringify(packageJson, null, 2));
+    // TODO: could add more templating.
+    await fs.writeFile(indexPath, indexIdyll.replace(/\{\{title\}\}/g, title));
+  }
+
+  async function installDependencies () {
+    return new Promise((resolve, reject) => {
+      let installer = spawn('yarn', ['install'], {
```

Comment:
Can we check for the existence of yarn first? The current behavior is to fall back to NPM if it isn't available... or maybe we just use npm

Reply:
Hmm, it's probably best to stay consistent across users in this case, even though it shouldn't make a difference. I've run into parity issues before -- updated to just use npm